File: Sum-of-Two-Values.py
# Author : ysh
# 2025/03/30 Sun 23:52:21
# https://cses.fi/problemset/task/1640
# TLE

import sys

input = sys.stdin.readline

a, b = [int(i) for i in input().split()]

pre = {}
t = 0
for i in [int(i) for i in input().split()]:
    ans = pre.get(b - i, None)
    # print(b - i, ans)
    if ans is not None:
        print(ans + 1, t + 1)
        quit()
    pre[i] = t
    t = t + 1

print('IMPOSSIBLE')
quit()

File: Sum-of-Two-Values.cpp
// Author : ysh
// 2025/03/30 Sun 23:38:32
// https://cses.fi/problemset/task/1640
#include<bits/stdc++.h>
using namespace std;
#include<slow>
int main() {
    ios::sync_with_stdio(false);
    cin.tie(0);

    int a,b;cin>>a>>b;
    vc<pair<int,int>>f(a);
    
    int t = 0;
    repo(&i, f) cin>>i.first, i.second = t++;

    sort(all(f));
    int l, r;l = 0, r = a - 1;
    while(l < r) {
        if(f.at(l).first + f.at(r).first == b) {
            outs(f.at(l).second + 1);
            out(f.at(r).second + 1);
            return 0;
        }

        if(r >= 0 && f.at(l).first + f.at(r).first > b) r--;
        else if(l < a && f.at(l).first + f.at(r).first < b) l++;
    }

    outt(IMPOSSIBLE);
    return 0;
}